project/src/main/kotlin/org/ionproject/core/klass/KlassController.kt

```diff
@@ -0,0 +1,31 @@
+package org.ionproject.core.klass
+
+import org.ionproject.core.common.SIREN_MEDIA_TYPE
+import org.ionproject.core.common.Siren
+import org.ionproject.core.common.Uri
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.*
+
+@RestController
+class KlassController(private val repo: KlassRepo) {
+
+    @ExceptionHandler
+    fun handleClassNotInDbException(exception: ClassNotInDbException) =
+        ResponseEntity.notFound().build<Any>()
+
+    @GetMapping(Uri.klassByTerm, produces = [SIREN_MEDIA_TYPE])
+    fun get(@PathVariable cid: Int, @PathVariable calterm: String): Siren {
+        val klass: FullKlass = repo.get(cid, calterm)
+
+        return KlassToSiren.toSiren(klass)
+    }
+
+    @GetMapping(Uri.klasses, produces = [SIREN_MEDIA_TYPE])
+    fun getCollection(@PathVariable cid: Int,
+                      @RequestParam(defaultValue = "0") page: Int,
+                      @RequestParam(defaultValue = "5") size: Int): Siren {
```

Comment:
If I made a request to get the classes without any query parameters, I would expect to return everything and not limited by 5. (More of an opinion, to further debate...)

Reply:
I think we need to just define a default value in the read API